x/nexus/keeper/general_message.go

```diff
@@ -198,13 +199,16 @@ func (k Keeper) GetSentMessages(ctx sdk.Context, chain exported.ChainName, limit
 		CountTotal: false,
 		Reverse:    false,
 	}
+	keyPrefix := append(sentMessagePrefix.Append(key.From(chain)).Bytes(), []byte(key.DefaultDelimiter)...)
 
-	query.Paginate(prefix.NewStore(k.getStore(ctx).KVStore, append(sentMessagePrefix.Append(key.From(chain)).Bytes(), []byte(key.DefaultDelimiter)...)), pageRequest, func(key []byte, value []byte) error {
+	if _, err := query.Paginate(prefix.NewStore(k.getStore(ctx).KVStore, keyPrefix), pageRequest, func(key []byte, value []byte) error {
 		ids = append(ids, string(value))
 		return nil
-	})
+	}); err != nil {
+		return nil, err
+	}
```

Comment:
I'd rather panic here instead of panicking at every call site, when it is expected that this never fails

Reply:
Yeah, I just don't like coupling the IAVL behaviour in the app, so I was keeping the GetSentMessage API consistent to make it easy to change this later. But I prefer this over doing another RunCached in the abci lol

Reply:
Yeah, I just don't like coupling the IAVL behaviour in the app, so I was keeping the GetSentMessage API consistent to make it easy to change this later. But I prefer this over doing another RunCached in the abci lol